GitUI/BranchTreePanel/RepoObjectsTree.Nodes.cs

```diff
@@ -218,8 +219,37 @@ public virtual string DisplayText()
                 return ToString();
             }
 
+            protected void SetNodeFont(FontStyle style)
+            {
+                if (style == FontStyle.Regular)
+                {
+                    // For regular, set to null to use the NativeTreeView font
+                    if (TreeViewNode.NodeFont != null)
+                    {
+                        TreeViewNode.NodeFont.Dispose();
+                        TreeViewNode.NodeFont = null;
+                    }
+                }
+                else
+                {
+                    // If current font doesn't have the input style, get rid of it
+                    if (TreeViewNode.NodeFont != null && !TreeViewNode.NodeFont.Style.HasFlag(style))
+                    {
+                        TreeViewNode.NodeFont.Dispose();
+                        TreeViewNode.NodeFont = null;
+                    }
+
+                    // If non-null, our font is already valid, otherwise create a new one
+                    if (TreeViewNode.NodeFont == null)
+                    {
+                        TreeViewNode.NodeFont = new Font(AppSettings.Font, style);
+                    }
+                }
+            }
```

Comment:
Can we simplify it as follows?
```
            protected void SetNodeFont(FontStyle style)
            {
                if (style != FontStyle.Regular && TreeViewNode.NodeFont.Style.HasFlag(style))
                {
                    return;
                }

                TreeViewNode.NodeFont.Dispose();
                TreeViewNode.NodeFont = null;
                TreeViewNode.NodeFont = new Font(AppSettings.Font, style);
            }
```

Any reason a node font would be `null`?

Reply:
We definitely want the NodeFont to be null by default so that it uses the font that was set on the NativeTreeView. This is the 99% case (or whatever), and in that case, we avoid allocating/disposing of node fonts. In the end, we only pay to dispose/allocate for the bold nodes. It's this optimization of making the nodes null for regular style that makes this code more complex, unfortunately.

It can be made a lot more simple at this level if we work on doing proper font management at the AppSettings level, like I did in the initial commit for this PR. But my change was flawed, and as per the long discussion with @vbjay, a system where we'd cache font + style + context with callbacks for when fonts change would probably be the way to go. For now, though, the change I provide here is optimal, behaves correctly, at the cost of a little complexity in the one function (SetNodeFont).

Reply:
Thank you for the explanation. Makes sense.